src/rpc.rs

```diff
@@ -573,8 +606,105 @@ impl RPC {
         }
     }
 
-    pub fn notify(&self) {
-        self.notification.send(Notification::Periodic).unwrap();
+    /// Attempt to notify scripthash subscriptions affected by transaction.
+    /// The `notified` hashset is an optimization to avoid double notifications.
+    /// When `notify_inputs` is set, we also notify scripthashes that are spent from.
+    fn try_notify_subscriptions_for_tx(
+        &self,
+        txid: &Sha256dHash,
+        mut blockhash: Option<Sha256dHash>,
+        notified: &mut HashSet<Sha256dHash>,
+        notify_inputs: bool,
+    ) -> Result<()> {
+        if notified.contains(txid) {
+            return Ok(());
+        }
+        notified.insert(txid.clone());
+
+        if blockhash.is_none() {
+            match self.query.lookup_confirmed_blockhash(txid, None) {
+                Ok(hash) => blockhash = hash,
+                Err(err) => trace!("Failed to lookup blockhash for {}: {}", txid, err),
+            };
+        }
+
+        let txn = self.query.load_txn(txid, blockhash)?;
+
+        let scripthashes: Vec<FullHash> = txn
+            .output
+            .iter()
+            .map(|o| compute_script_hash(&o.script_pubkey[..]))
+            .collect();
+
+        for s in scripthashes {
+            if let Err(e) = self.notification.send(Notification::ScriptHashChange(s)) {
+                trace!("ScriptHash change notification failed {}", e);
+            }
+        }
+
+        if notify_inputs {
+            for txin in txn.input {
+                if txin.previous_output.is_null() {
+                    continue;
+                }
+                let id: &Sha256dHash = &txin.previous_output.txid;
+
+                if let Err(e) = self.try_notify_subscriptions_for_tx(id, None, notified, false) {
```

Comment:
You are calling `try_notify_subscriptions_for_tx` for a new transaction's spending transactions.
Recall that you are parsing the script-hashes of all the outputs in this function (line 633).
So I believe you unintentionally send notifications for sibling outputs of the spent transaction (instead of focusing only on the relevant output which was spent in the new transaction).

Reply:
Good observation